pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -327,21 +327,29 @@ func (p *K8SPlugin) getContainerIDFromCGroups(pid int32) (string, error) {
 	}
 
 	for _, cgroup := range cgroups {
-		// We are only interested in kube pods entries. Example entry:
+		// We are only interested in kube pods entries. Example entries:
 		// 11:hugetlb:/kubepods/burstable/pod2c48913c-b29f-11e7-9350-020968147796/9bca8d63d5fa610783847915bcff0ecac1273e5b4bed3f6fa1b07350e0135961
-		if len(cgroup.GroupPath) < 9 {
+		// 12:pids:/docker/8d461fa5765781bcf5f7eb192f101bc3103d4b932e26236f43feecfa20664f96/kubepods/besteffort/poddaa5c7ee-3484-4533-af39-3591564fd03e/aff34703e5e1f89443e9a1bffcc80f43f74d4808a2dd22c8f88c08547b323934
+		groupPath := cgroup.GroupPath
+		if len(groupPath) < 9 {
```

Comment:
this length check is now unnecessary (it existed to prevent slicing out of bounds to create `substring`)

Reply:
I fixed it with https://github.com/spiffe/spire/pull/1133/commits/f9ab8e546c643fb3f39f86751ffb65c351eafd01 .